thread_runner.py

```diff
@@ -30,7 +30,22 @@ def waitFor(self):
     def runLoopedFunction(self):
         while True:
             if self.thread_running:
-                self.output = self.func(**self.args)
+                total_tries = 3
+                current_try = 0
```

Comment:
You could also have a tries_left that you decrement each time and inizialize to 3